src/network/networkd-dhcp4.c

```diff
@@ -1163,6 +1165,19 @@ int dhcp4_configure(Link *link) {
                         return log_link_error_errno(link, r, "DHCP4 CLIENT: Failed to set request flag for timezone: %m");
         }
 
+        SET_FOREACH(request_options, link->network->dhcp_request_options, i) {
+                uint32_t option = PTR_TO_UINT32(request_options);
+
+                r = sd_dhcp_client_set_request_option(link->dhcp_client, option);
+                if (r == -EEXIST) {
+                        log_link_debug(link, "DHCP4 CLIENT: Failed to set request flag for '%u', ignoring.", option);
```

Comment:
> Would debug msgs show by default, is there a warn? Maybe mention that the option is already set?
> 

Yes .
> I wonder too if there are scenarios where a previously set option may want to be overridden.
> 

By default networkd add request options which may this option can duplicate hence this scenario.

> I noticed too that sd_dhcp_client_set_request_option has a switch to prevent specific options from being set...
> 
yes why would pad overload such thing allowed to be set ?

> And also rfc951 mentions a magic number which makes the option order meaningful, don't know that anyone cares about that though? "If the 'vend' field is used, it is recommended that a 4 byte 'magic number' be the first item within 'vend'."

Is that a request option ? I think that is a option (64) to send with data such as user specific

Reply:
Parameter list was one thing I was hoping to modify, since it can be used to profile devices. It would already be set too I imagine in some cases so kind of a complicated situation. I was hoping to match this other devices request options, even though I UseDNS=false for example

Reply:
you can disable theme via https://www.freedesktop.org/software/systemd/man/systemd.network.html#Anonymize= 

But if you want to send some data please open an another RFE

Reply:
Ok. Was hoping to send a "profile" (send a request parameters option) that matched this other clients request-- even though it doesn't match up. Its just one of the options that sd_dhcp_client_set_request_option won't let you set